# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/GCAPI.h
# Commit: 662cb63c42d7
# Full Hash: 662cb63c42d7874a79256a6edf1a03d83e43a014
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 2
# Description:
#   Bug 1692065 - wasm: Create a unique wasm::AnyRef type. r=yury,jonco
#   
#   Right now wasm references area represented as a JSObject*. This is assumed in
#   many places in SpiderMonkey, most prominently in our GC integration
#   around tracing, the store buffer, and write barriers. This will cause issues
# ==============================================================================

diff -r 8b506ab41451 -r 662cb63c42d7 js/public/GCAPI.h
--- a/js/public/GCAPI.h	Fri Aug 04 13:54:57 2023 +0000
+++ b/js/public/GCAPI.h	Fri Aug 04 14:06:49 2023 +0000
@@ -603,7 +603,7 @@
   D(DISABLE_GENERATIONAL_GC, 24)                                       \
   D(FINISH_GC, 25)                                                     \
   D(PREPARE_FOR_TRACING, 26)                                           \
-  D(UNUSED4, 27)                                                       \
+  D(FULL_WASM_ANYREF_BUFFER, 27)                                       \
   D(FULL_CELL_PTR_STR_BUFFER, 28)                                      \
   D(TOO_MUCH_JIT_CODE, 29)                                             \
   D(FULL_CELL_PTR_BIGINT_BUFFER, 30)                                   \